File: backend/app/schemas/audit.py
"""Audit log schemas."""
from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: str
    organization_id: str
    actor_id: str
    actor_name: str | None = None
    actor_email: str | None = None
    action: str
    resource_type: str
    resource_id: str | None = None
    resource_name: str | None = None
    details: dict[str, Any] | None = None
    ip_address: str | None = None
    user_agent: str | None = None
    created_at: str

File: frontend/src/types/audit.ts
/** Audit log types */
export interface AuditLogEntry {
  id: string
  organization_id: string
  actor_id: string
  actor_name?: string | null
  actor_email?: string | null
  action: string
  resource_type: string
  resource_id?: string | null
  resource_name?: string | null
  details?: Record<string, any> | null
  ip_address?: string | null
  user_agent?: string | null
  created_at: string
}
